### DISCUSSION TITLE
[V5] Valid GET routes return 404 while using HEAD request

### DISCUSSION THREAD

**Discussion by danpospisil** [adonisjs/core#2136] [Help] [answered]
I need my valid GET routes to return 200 on HEAD request. Fresh Adonis install returns 200 on GET but 404 on HEAD request on the same route. How to reproduce:

1. install fresh V5
2. start the server
3. issue GET request: `curl -s -o /dev/null -w "%{http_code}" 127.0.0.1:3333` -> returns 200
4. issue HEAD request: `curl -I -s -o /dev/null -w "%{http_code}" 127.0.0.1:3333` -> returns 404

I tried to search the docs but I haven't found the way how to enable HEAD requests. 
Thanks

**Comment by danpospisil** [comment, marked as answer]
In v4.1 GET routes have HEAD as well:
https://github.com/adonisjs/core/blob/2ec8e54d3ad02ae215f24ee9b577a835dff5b7a8/src/Route/Manager.js#L143
https://github.com/adonisjs/core/blob/2ec8e54d3ad02ae215f24ee9b577a835dff5b7a8/src/Route/Brisk.js#L60

But in v5 the HEAD was removed:
https://github.com/adonisjs/http-server/blob/ea55c2a65fd388373d0b4e35ae45bee9cb096d02/src/Router/index.ts#L143
https://github.com/adonisjs/http-server/blob/ea55c2a65fd388373d0b4e35ae45bee9cb096d02/src/Router/BriskRoute.ts#L61

When I add it back, it starts to work.

Is there any reason for this @thetutlage ?

**Reply by thetutlage** [reply]
I think we should add it back. Can you please send a PR for the same?

**Reply by danpospisil** [reply]
@thetutlage - done: https://github.com/adonisjs/http-server/pull/32 I have also added it to Resource routes and modified all necessary tests as well
